docs/src/main/paradox/server-class.md

```diff
@@ -0,0 +1,117 @@
+
+The remaining directives used for this route are:
+
+* `~` : fuses `Route`s together - this will become more apparent when you see the complete `Route` definition here below.
+* `delete` : matches against the Http directive `DELETE`.
+
+The "business logic" for when deleting a user is straight forward; send an instruction about removing a user to the user registry actor and return a status code to the client (which in this case is `StatusCodes.OK`, i.e. Http status code 200)
+
+### Retrieving all users
+
+Finally we should implement functionality to retrieve all registered users:
+
+@@snip [QuickstartServer.scala]($g8src$/scala/com/lightbend/akka/http/sample/QuickstartServer.scala) { #users-get }
+
+This code is based on the same structure as the code above, send a message to the user registry actor using an `ask` and pass on the `Future` to the `complete` method.
+
+Why do we not use an `onComplete` as we did for when retrieving a particular user? The difference is that when we use `GetUsers` there will always be something returned; an empty list means that there are no registered users. However, when we asked for a particular user ID, there might be the case that there is no such user recorded and we need a way to tell this to the client. When sent a `GetUser(name)` the user registry actor therefore send back an `Option[UserInfo]`. It could be that there is a `Some(UserInfo)` or a `None` which indicates that there was no match in the registry.
+
+### The complete Route
+
+Below is the complete `Route` definition used in the sample application:
+
+@@snip [QuickstartServer.scala]($g8src$/scala/com/lightbend/akka/http/sample/QuickstartServer.scala) { #all-routes }
+
+So far we have referred to `Route` without explaining what it is but now is the time to do so. Under the hood, Akka HTTP uses [Akka Streams](http://doc.akka.io/docs/akka/current/scala/stream/index.html). We don't have time to cover Akka Streams here, but if you are interested, you should take a look at the Hello World sample application for Akka Streams. Since Akka HTTP is built on top of Akka Streams, it means that some concepts of Akka Streams are available for us to use. In the case of `Route` you can think of it as a flow of in- and outbound data which is a perfect fit for Akka Streams. (Technically the `Route` type is `RequestContext ⇒ Future[RouteResult]` but there is no need to worry about what that means now.)
+
+## Http server
+
+To set up an Akka HTTP server we must first define some implicit values that will be used by the server:
+
+@@snip [QuickstartServer.scala]($g8src$/scala/com/lightbend/akka/http/sample/QuickstartServer.scala) { #server-bootstrapping }
+
+What does the above mean and why do we need it?
+
+* `ActorSystem` : the context in which actors will run. What actors, you may wonder? Akka Streams uses actors under the hood, and the actor system defined in this `val` will be picked up and used by Streams.
```

Comment:
The main reason to have an ActorSystem would be to run the service. Then we can also mention that Akka Http needs this as well.